https://issues.redhat.com/browse/ACM-13088 (#6770)
open-cluster-management/rhacm-docs: governance/gatekeeper_operator/manage_gk.adoc

[#managing-gk-policies]
= Managing the {gate} installation policies

Use the {acm-short} policy to install the {gate} and Gatekeeper on a managed cluster. 

*Required access*: Cluster administrator

To create, view, and update your {gate} installation policies, complete the following sections: 

- <<installing-gatekeeper-operator-policy,Installing Gatekeeper using a {gate} policy>>
- <<creating-a-gatekeeper-policy-from-the-console,Creating a Gatekeeper policy from the console>>
- <<upgrading-gatekeeper-gatekeeper-operator,Upgrading Gatekeeper and the {gate}>>
- <<disabling-gatekeeper-operator-policy,Disabling {gate} policy>>
- <<deleting-gatekeeper-operator-policy,Deleting {gate} policy>>

> - <<uninstalling-gatekeeper,Uninstalling Gatekeeper policy, Gatekeeper, and {gate} policy>>

Reviewer: These all need to be updated to match the new tags (these are also noted in the CI):
```
asciidoctor: INFO: possible invalid reference: installing-gatekeeper-operator-policy
asciidoctor: INFO: possible invalid reference: creating-a-gatekeeper-policy-from-the-console
asciidoctor: INFO: possible invalid reference: upgrading-gatekeeper-gatekeeper-operator
asciidoctor: INFO: possible invalid reference: disabling-gatekeeper-operator-policy
asciidoctor: INFO: possible invalid reference: deleting-gatekeeper-operator-policy
asciidoctor: INFO: possible invalid reference: uninstalling-gatekeeper
```
ref: https://github.com/stolostron/rhacm-docs/actions/runs/10888730503/job/30213839121?pr=6770#step:4:8

[#installing-gk-operator-policy]
== Installing Gatekeeper using a {gate} policy

> To install the {gate} policy, use the configuration policy controller. During the install, the operator group and subscription pull the {gate} to install it on your managed cluster. Then, the {gate} creates a Gatekeeper custom resource to configure Gatekeeper. 

Reviewer: ```suggestion
To install the {gate} policy, use the configuration policy controller. During the install, the operator group and subscription pull the {gate} to install it on your managed cluster. Then, the policy creates a Gatekeeper custom resource to configure Gatekeeper. 
```

> To view the status of the {gate} policy, see the {acm-short} configuration policy controller because it checks the {gate} policy and supports the  `enforce` remediation action. When you set the controller to `enforce` it automatically creates the {gate} objects on the managed cluster. 

Reviewer: These feel like extra words to me.
```suggestion
The {acm-short} configuration policy controller checks the {gate} policy and supports the  `enforce` remediation action. When you set the controller to `enforce` it automatically creates the {gate} objects on the managed cluster. 
```

[#creating-gk-operator-console]
== Creating a Gatekeeper policy from the console

> When you create a Gatekeeper policy from the console, you must set your remediation `enforce` to install Gatekeeper. By default, the operator generates the values. 

Reviewer: I'm not sure what this is getting at, so maybe just take it out?
```suggestion
When you create a Gatekeeper policy from the console, you must set your remediation `enforce` to install Gatekeeper.
```

Reviewer 2: We need a brief description for the section

Reviewer: I don't mean the whole paragraph--I mean the sentence "By default, the operator generates the values."

Reviewer 2: ohhh gotchu, sorry read through too quickly lol

[#viewing-gk-operator-policy]

> === Viewing the {gate} policy

Reviewer: Is it intentional to have this as a subheading of the "Creating" section? It's fine as it is--I'm just curious.

Reviewer 2: Yes this is intentional

To view your {gate} policy and its status from the console, complete the following steps:

. Select the `policy-gatekeeper-operator` policy to view more details.
. Select the _Clusters_ tab to view the policy violations. 

[#upgrading-gk-and-gk-operator]
== Upgrading Gatekeeper and the {gate}

You can upgrade the versions for Gatekeeper and the {gate}. When you install the {gate} with the {gate} policy, notice the value for `upgradeApproval`. The operator upgrades automatically when you set `upgradeApproval` to `Automatic`.

> If you set `upgradeApproval` to `Manual`, you must manually approve the upgrade for each cluster of the {gate}.

Reviewer: ```suggestion
If you set `upgradeApproval` to `Manual`, you must manually approve the upgrade for each cluster where the {gate} is installed.
```

[#disabling-gk-operator-policy]
== Disabling {gate} policy

> To disable your `policy-gatekeeper-operator` policy, select the *Disable* option from the _Actions_ menu.

Reviewer: ```suggestion
To disable your `policy-gatekeeper-operator` policy, select the *Disable* option from the _Actions_ menu in the console, or set `spec.disabled: true` from the CLI.
```

[#deleting-gk-operator-policy]
== Deleting {gate} policy

To delete your {gate} policy from your CLI, complete the following steps: 

. Delete {gate} policy by running the following command:

+
[source,bash]
----
oc delete policies.policy.open-cluster-management.io <policy-gatekeeper-operator-name> -n <namespace>
----

 . Verify that you deleted your policy by running the following command:

+
[source,bash]
----
oc get policies.policy.open-cluster-management.io <policy-gatekeeper-operator-name> -n <namespace>
----

To delete your {gate} policy from the console, click the *Actions* icon for the `policy-gatekeeper-operator` policy and select *Delete*.

[#uninstalling-gk]
== Uninstalling Gatekeeper constraint, Gatekeeper instance, and {gate} policy

To uninstall Gatekeeper policy, complete the steps in the following sections:

- <<removing-gatekeeper-constraint,Removing Gatekeeper _Constraint_>>
- <<removing-gatekeeper-instance,Removing Gatekeeper instance>>

> - <<removing-gatekeeper-operator,Removing {gate}>>

Reviewer: Not sure if you wanted these to also be `gk` rather than `gatekeeper`?

Reviewer 2: If the anchors are updated for these sections to have gk, we should

[#removing-gatekeeper-constraint]

> === Removing Gatekeeper _Constraint_

Reviewer: It's nuanced, but `ConstraintTemplate` is a kind, but constraint is not.
```suggestion
=== Removing Gatekeeper constraints
```

To remove the Gatekeeper constraint and `ConstraintTemplate` from your managed cluster, complete the following steps:

. Edit your Gatekeeper constraint or `ConstraintTemplate` policy. 
. Locate the template that you used to create the Gatekeeper `Constraint` and `ConstraintTemplate`.
. Delete the entries from the list of templates. (Or delete the policy if they're the only templates.)

> . Save and apply the policy.

Reviewer: I think a clarification here would be good.
```suggestion
. Save and apply the policy.

**Note:** This assumes the constraint and `ConstraintTemplate` are provided directly under `policy-templates` and are not wrapped in a `ConfigurationPolicy`.
```

Reviewer 2: Reminder about how we label notes. Also avoid directional terms like "under". What does it mean to be "wrapped"? Included? Reminder that this should be in present tense @jc-berger I would also remove "This assumes"

Reviewer: So something like this?

>**Note:** In these steps, the constraint and `ConstraintTemplate` are provided directly as items of `spec.policy-templates` in the `Policy` and are not contained in a `ConfigurationPolicy`.

[#removing-gatekeeper-instance]
=== Removing Gatekeeper instance 

To remove the Gatekeeper instance from your managed cluster, complete the following steps:

. Edit your {gate} policy. 
. Locate the `ConfigurationPolicy` template that you used to create the {gate} custom resource.

> . Change the value for `complianceType` of the `ConfigurationPolicy` template to `mustnothave`.

Reviewer: ```suggestion
. Change the value for `complianceType` of the `ConfigurationPolicy` template to `mustnothave`.
. This deletes the {gate} custom resource, which signals the {gate} to clean up the Gatekeeper deployment.
```

Reviewer 2: Does this need to be another step? Reminder to maintain present tense

Reviewer: Updated my comment to present tense (and added some clarifications). There could be an additional step like `Check that the policy status is "Compliant"`, but I'm not sure that's necessary--at least we didn't do that for the other steps here.

[#managing-gk-policies]
= Managing the {gate} installation policies

Use the {acm-short} policy to install the {gate} and Gatekeeper on a managed cluster. 

*Required access*: Cluster administrator

To create, view, and update your {gate} installation policies, complete the following sections: 

- <<installing-gk-operator-policy,Installing Gatekeeper using a {gate} policy>>
- <<creating-gk-operator-console,Creating a Gatekeeper policy from the console>>
- <<upgrading-gk-and-gk-operator,Upgrading Gatekeeper and the {gate}>>
- <<disabling-gk-operator-policy,Disabling {gate} policy>>
- <<deleting-gk-operator-policy,Deleting {gate} policy>>

> - <<uninstalling-gk,Uninstalling Gatekeeper policy, Gatekeeper, and {gate} policy>>

Reviewer: Consistent with the heading
```suggestion
- <<uninstalling-gk,Uninstalling Gatekeeper constraints, Gatekeeper instance, and {gate} policy>>
```

If you set `upgradeApproval` to `Manual`, you must manually approve the upgrade for each cluster where the {gate} is installed.

[#disabling-gk-operator-policy]
== Disabling {gate} policy

To disable your `policy-gatekeeper-operator` policy, select the *Disable* option from the _Actions_ menu in the console, or set `spec.disabled: true` from the CLI.

[#deleting-gk-operator-policy]
== Deleting {gate} policy

To delete your {gate} policy from your CLI, complete the following steps: 

. Delete {gate} policy by running the following command:

+
[source,bash]
----
oc delete policies.policy.open-cluster-management.io <policy-gatekeeper-operator-name> -n <namespace>
----

 . Verify that you deleted your policy by running the following command:

+
[source,bash]
----
oc get policies.policy.open-cluster-management.io <policy-gatekeeper-operator-name> -n <namespace>
----

To delete your {gate} policy from the console, click the *Actions* icon for the `policy-gatekeeper-operator` policy and select *Delete*.

[#uninstalling-gk]

> == Uninstalling Gatekeeper constraint, Gatekeeper instance, and {gate} policy

Reviewer: ```suggestion
== Uninstalling Gatekeeper constraints, Gatekeeper instance, and {gate} policy
```

To uninstall Gatekeeper policy, complete the steps in the following sections:

> - <<removing-gk-constraint,Removing Gatekeeper _Constraint_>>

Reviewer: ```suggestion
- <<removing-gk-constraint,Removing Gatekeeper constraints>>
```

- <<removing-gk-instance,Removing Gatekeeper instance>>
- <<removing-gk-operator,Removing {gate}>>

[#removing-gk-constraint]

> === Removing Gatekeeper _Constraints_

Reviewer: ```suggestion
=== Removing Gatekeeper constraints
```

To remove the Gatekeeper constraint and `ConstraintTemplate` from your managed cluster, complete the following steps:

. Edit your Gatekeeper constraint or `ConstraintTemplate` policy. 
. Locate the template that you used to create the Gatekeeper `Constraint` and `ConstraintTemplate`.
. Delete the entries from the list of templates. (Or delete the policy if they're the only templates.)
. Save and apply the policy.

> *Norw:* The constraint and `ConstraintTemplate` are provided in the `policy-templates` instead of the `ConfigurationPolicy`.

Reviewer: ```suggestion
*Note:* The constraint and `ConstraintTemplate` are provided directly in the `policy-templates` instead of inside of a `ConfigurationPolicy`.
```

Reviewer 2: ...instead of within a `ConfigurationPolicy`
